content/develop/start/hardhat.md

```diff
@@ -3,3 +3,241 @@ title: "Aurora: Getting Started with Hardhat"
+  const watermelon = watermelonToken.attach(token)
+  const [minter] = await ethers.getSigners();
+  const balance = (await (await watermelon.connect(minter)).balanceOf(account)).toNumber()
+  console.log(`Account ${account} has a total token balance:  ${balance} WTM`);
+});
+```
+
+To get the `balance`, use the following command:
+
+```bash
+$ npx hardhat balanceOf --token 0xD7f2A76F5DA173043E6c61a0A18D835809A07766 --account 0x6A33382de9f73B846878a57500d055B981229ac4 --network testnet_aurora
+Account 0x6A33382de9f73B846878a57500d055B981229ac4 has a total token balance:  999970 WTM
+```
+
+### Approve ERC20
+
+In some cases, instead of calling the `transfer` directly, the sender
+can approve a specific amount of tokens to be withdrawn from his account
+to specific receipient address later. This can be done by calling `approve`
+then calling `transferFrom`.
+
+```javascript
+task("approve", "ERC20 approve")
+    .addParam("token", "Token address")
+    .addParam("spender", "Spender address")
+    .addParam("amount", "Token amount")
+    .setAction(async function ({ token, spender, amount }, { ethers: { getSigners } }, runSuper) {
+        const watermelonToken = await ethers.getContractFactory("WatermelonToken")
+        const watermelon = watermelonToken.attach(token)
+        const [sender] = await ethers.getSigners();
+        await (await watermelon.connect(sender).approve(spender, amount)).wait()
+        console.log(`${sender.address} has approved ${amount} tokens to ${spender}`);
+    });
+
+module.exports = {};
+```
+
+To call `approve`, use the following command:
+
+```bash
+npx hardhat approve --token 0xD7f2A76F5DA173043E6c61a0A18D835809A07766 --spender 0x8722C88e82AbCC639148Ab6128Cd63333B2Ad771 --amount 10 --network testnet_aurora
+0x6A33382de9f73B846878a57500d055B981229ac4 has approved 10 tokens to 0x8722C88e82AbCC639148Ab6128Cd63333B2Ad771
+```
+
+### TransferFrom ERC20
+
+After approving the tokens, a receipient can call `transferFrom` to move
+the `allowance` to his account.  
+
+```javascript
+task("transferFrom", "ERC20 transferFrom")
+    .addParam("token", "Token address")
+    .addParam("sender", "Sender address")
+    .addParam("amount", "Token amount")
+    .setAction(async function ({ token, sender, amount }, { ethers: { getSigners } }, runSuper) {
+        const watermelonToken = await ethers.getContractFactory("WatermelonToken")
+        const watermelon = watermelonToken.attach(token)
+        const [recipient] = await ethers.getSigners()
+        console.log(recipient.address);
+        await (await watermelon.connect(recipient).transferFrom(sender, recipient.address, amount)).wait()
+        console.log(`${recipient.address} has received ${amount} tokens from ${sender}`)
+    });
+```
+
+To call `transferFrom`, use the following command:
+
+```bash
+npx hardhat transferFrom --token 0xD7f2A76F5DA173043E6c61a0A18D835809A07766 --sender 0x6A33382de9f73B846878a57500d055B981229ac4  --amount 10 --network testnet_aurora
```

Comment:
How does this command know it's supposed to come from `0x8722C88e82AbCC639148Ab6128Cd63333B2Ad771`?

Reply:
Nice catch. I forgot to add `export the address private key of that address`.
Because in the ERC20 contract it gets the `msg.sender` as the recipient.